src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs

```diff
@@ -173,6 +173,99 @@ private static bool IsInvalidDouble(double number)
         {
             return StandardSingleColumnTable<T>((runner, symbolContext, irContext, args) => targetFunction(irContext, args));
         }
+
+        public static Func<EvalVisitor, SymbolContext, IRContext, FormulaValue[], FormulaValue> DoubleSingleColumnTable<T>(Func<EvalVisitor, SymbolContext, IRContext, T[], FormulaValue> targetFunction) where T : FormulaValue
+        {
+            return (runner, symbolContext, irContext, args) =>
+            {
+                var arg0 = args[0];
+                var arg1 = args[1];
+
+                KeyValuePair<IEnumerable<DValue<RecordValue>>, IEnumerable<DValue<RecordValue>>> allRows;
+                if (arg0 is TableValue both0 && arg1 is TableValue both1)
+                {
+                    if (both0.Rows.Count() == both1.Rows.Count())
+                    {
+                        allRows = new KeyValuePair<IEnumerable<DValue<RecordValue>>, IEnumerable<DValue<RecordValue>>>(both0.Rows, both1.Rows);
+                    }
+                    else
+                    {
+                        return CommonErrors.UnreachableCodeError(irContext);
```

Comment:
We can have tables of different lengths being concatenated. The existing behavior is to complete the smaller table with blank / null values:
![image](https://user-images.githubusercontent.com/44986255/142940584-5a00d3ed-44c0-4234-bebc-1baacd10ddab.png)

Reply:
This is astonishing! Thanks for pointing this out, this is a major oversight in the current code.

Reply:
Added support for variable length input tables.